File: common/src/main/java/com/rongcloud/common/utils/JsonUtils.java
/*
 * Copyright © 2021 RongCloud. All rights reserved.
 */

package com.rongcloud.common.utils;

import com.google.gson.Gson;

/**
 * @author gusd
 * @Date 2021/06/04
 */
public class JsonUtils {
    private static final Gson gson;

    static {
        gson = new Gson();
    }

    public static String toJson(Object object) {
        return gson.toJson(object);
    }

    public static <T> T fromJson(String jsonString, Class<T> clazz) {
        return gson.fromJson(jsonString, clazz);
    }
}

File: app/src/main/java/cn/rongcloud/voiceroomdemo/utils/JsonUtils.kt
/*
 * Copyright © 2021 RongCloud. All rights reserved.
 */

package cn.rongcloud.voiceroomdemo.utils

import com.google.gson.Gson

/**
 * @author gusd
 * @Date 2021/06/08
 */
object JsonUtils {
    private val gson: Gson by lazy {
        Gson()
    }

    fun toJson(obj: Any): String {
        return gson.toJson(obj)
    }

    fun <T> fromJson(jsonString: String, clazz: Class<T>): T? {
        return gson.fromJson(jsonString, clazz)
    }
}